sympy/tensor/array/tests/test_array_comprehension.py

```diff
@@ -73,5 +73,7 @@ def test_arraycomprehensionmap():
     # tests about lambda expression
     assert ArrayComprehensionMap(lambda: 3, (i, 1, 5)).doit().tolist() == [3, 3, 3, 3, 3]
     assert ArrayComprehensionMap(lambda i: i+1, (i, 1, 5)).doit().tolist() == [2, 3, 4, 5, 6]
-    raises(ValueError, lambda: ArrayComprehensionMap(lambda i, j: i+j, (i, 1, 5)).doit())
     raises(ValueError, lambda: ArrayComprehensionMap(i*j, (i, 1, 3), (j, 2, 4)))
+    with warns_deprecated_sympy():
+        a = ArrayComprehensionMap(lambda i, j: i+j, (i, 1, 5))
+        raises(ValueError, lambda: a.doit())
```

Comment:
I think this is fine but note that there is also `ignore_warnings` rather than `warns_deprecated_sympy`. The intention of `ignore_warnings` is simply to hide any warnings rather than to assert that there are warnings.